ldms/src/ldmsd/ldmsd.h

```diff
@@ -931,6 +931,7 @@ typedef struct ldmsd_plugin {
 	int (*config)(struct ldmsd_plugin *self, struct attr_value_list *kwl, struct attr_value_list *avl);
 	void (*term)(struct ldmsd_plugin *self);
 	const char *(*usage)(struct ldmsd_plugin *self);
+	ldmsd_cfgobj_t cfgobj; /* cfgobj associated to instance */
```

Comment:
@tom95858 This is a thing that I don't think we should just land. This moves us in the wrong direction. We need to be _removing_ ldmsd's internal state from the plugin's API struct, not adding more. This is where we're currently failing to have good abstraction, encapsulation, and laying in the plugin API.

In my code (which granted, is still WIP) already remove _all_ of ldmsd's internal state from that struct. This is going to introduce even more that I'll have to work to back out.

I think we should remove this commit from this PR.

Reply:
Removed...I think it was detritus.

Reply:
I _think_ it was added solely to give json_stream_sampler access to the ref in the cfgobj. But yes, I didn't immediately see how that ref would be necessary. I reverted those commits in my last rebase too (and made a note to myself to revisit them later when I figured out why the ref get/put was happening in json_stream_handler). But if you don't see a reason for it either, it makes things easier to just get rid of it. Thanks!